src/FixedDataTable.js

```diff
@@ -1056,6 +1056,19 @@ var FixedDataTable = React.createClass({
 
     this._scrollHelper.setViewportHeight(bodyHeight);
 
+    // This calculation is synonymous to Element.scrollTop
+    var scrollTop = firstRowIndex != null ? Math.abs(firstRowOffset - this._scrollHelper.getRowPosition(firstRowIndex)) : 0;
+    // Handle the case where the scrollTop is beyond the maxScrollY, such as when the user is completely scrolled
+    // down and resizes the viewport to be smaller vertically. The other case is when resizing the viewport large enough
+    // so that a scrollbar is not necessary anymore, to make sure we set the scrollTop back to 0.
+    if (scrollTop > maxScrollY || (oldState && oldState.scrollY !== 0 && maxScrollY === 0)) {
```

Comment:
Use scrollY instead of oldState.scrollY. We might have modified it by the time we reach this line